pkg/database/import.go

```diff
@@ -0,0 +1,158 @@
+package database
+
+import (
+	"github.com/rotisserie/eris"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+// Import will copy the contents of input db to output db.
+func Import(input, output *DbInstance, dryRun bool) error {
+	in := input.DB
+	if err := output.DB.Transaction(func(out *gorm.DB) error {
+		if err := importTable(in, out, dryRun, Experiment{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, ExperimentTag{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Run{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Tag{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Metric{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, LatestMetric{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Param{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, Dashboard{}); err != nil {
+			return err
+		}
+		if err := importTable(in, out, dryRun, App{}); err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		return eris.Wrapf(err, "error importing table data")
+	}
+
+	return nil
+}
+
+// importTable will copy the contents of one table (model) from sourceDB to destDB.
+func importTable[T any](sourceDB, destDB *gorm.DB, dryRun bool, model T) error {
+	// Query data from the source database
+	log.Infof("Importing %T", model)
+	rows, err := sourceDB.Model(&model).Rows()
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	log.Infof("Transferring records (dry run? %v)", dryRun)
+
+	// Stream data to the destination database in bulk with collision handling
+	for rows.Next() {
+		var item T
+		sourceDB.ScanRows(rows, &item)
+		found, err := findCollision(destDB, item)
+		if err != nil {
+			return err
+		}
+		if found {
+			// Handle the collision with a warning
+			log.Infof(
+				`Skipping record (type/values) '%T/%v', already present in dest`,
+				item,
+				item,
+			)
+		} else {
+			if !dryRun {
+				destDB.Create(&item)
```

Comment:
we probably still want to batch those inserts (probably a configurable amount, defaulting to 100?)

Reply:
I think it will add some complexity since we have to accumulate up to the batch size across that number of iterations but I can try it

Reply:
using transactions per table for now